src/components/App.component.js

```diff
@@ -339,17 +346,43 @@ class App extends React.Component {
                     open={!!this.state.snackbar}
                     style={styles.snackbar}
                 />
-                <div className="content-area">
-                    {this.state.appSearch
-                        ? this.renderSearchResults()
-                        : this.renderSection(
-                              this.state.section,
-                              this.state.installedApps,
-                              true
-                          )}
-                    {this.renderUploadProgress()}
-                    {this.renderInstallProgress()}
-                </div>
+                <DataQuery query={bundledAppsQuery}>
+                    {({ data }) => {
+                        const bundledApps = (data
+                            ? data.bundledApps.modules
+                            : []
+                        )
+                            .filter(
+                                app =>
+                                    !this.state.installedApps.find(
+                                        a =>
+                                            a.isBundledApp &&
+                                            a.short_name === app.name
+                                    )
+                            )
```

Comment:
It seems to me that this will only return the bundled apps which have already been "installed" (overrided) in the App Hub, is that correct?  If so, I don't think we need to make the `getModules.action` call at all, couldn't we just filter the `installedApps` list by `isBundled`?

Also - `isBundledApp` is incorrect here, it was changed to `.bundled` - see https://play.dhis2.org/dev/api/apps

Reply:
The issue with `installedApps` is that its initial state is from `d2.system.installedApps`, which only lists installed/overriden apps from the App Hub. On the other hand, `getModules` returns all available apps. The purpose of this filter is to allow us to [create a list of all apps (both installed and bundled) without duplicates](https://github.com/dhis2/app-management-app/pull/181/files#diff-49a6b774cf5156e1bfecf0437f141234b3f52913be1a57b5a42c42f211c7eb98R370-R373) and allow users to update apps that have not been overriden.